### ISSUE TITLE
[CMake] It's 2020, where is CMake?

### ISSUE THREAD

**Issue by a9448f16-e2da-41ea-9cdf-3a0d7ad482e3** [python/cpython#84920]
BPO | [40743](https://bugs.python.org/issue40743)
--- | :---
Nosy | @tiran, @ned-deily, @pmp-p, @cjerdonek, @remilapeyre

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2020-06-07.03:41:21.111>
created_at = <Date 2020-05-23.14:56:08.729>
labels = ['type-feature', '3.10', 'build']
title = "[CMake] It's 2020, where is CMake?"
updated_at = <Date 2020-06-07.03:41:21.107>
user = 'https://bugs.python.org/TheUltimatePineapple'
```

bugs.python.org fields:
```python
activity = <Date 2020-06-07.03:41:21.107>
actor = 'ned.deily'
assignee = 'none'
closed = True
closed_date = <Date 2020-06-07.03:41:21.111>
closer = 'ned.deily'
components = ['Build']
creation = <Date 2020-05-23.14:56:08.729>
creator = 'TheUltimatePineapple'
dependencies = []
files = []
hgrepos = []
issue_num = 40743
keywords = []
message_count = 7.0
messages = ['369724', '369725', '369733', '369738', '369740', '370866', '370871']
nosy_count = 6.0
nosy_names = ['christian.heimes', 'ned.deily', 'pmpp', 'chris.jerdonek', 'remi.lapeyre', 'TheUltimatePineapple']
pr_nums = []
priority = 'normal'
resolution = 'postponed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue40743'
versions = ['Python 3.10']
```

</p></details>

**Comment by ResearchDaniel** [comment]
As a side note to point 1 it is now possible to include Python into a CMake-based project using the vcpkg package manager. As it is quite tricky to get isolated mode working for Windows, MacOS, and Linux I made an example:
https://github.com/ResearchDaniel/python3-embedded-example

**Comment by v-fox** [comment]
It's astonishing how easily and aggressively this guy was dismissed.

As a user of Gentoo I can assuredly say that autotools is a major source of pain. Mainly because for all non-large builds it's slower than compilation itself. Especially in combination with portage that is written in python which is slow by itself due to current lack of JIT and proper multi-threading (portage doesn't support non-locked builds and JIT breaks installation phase completely on python-3.13/LLVM-18).

Both autotools and python are dicking around in their bash/python scripts for ages (try running `emerge -v1a -uD --keep-going @world` for update when having over 2000 packages), it's comically embarrassing for modern hardware.
If you're using gcc by default then its comically slow LTO is a cherry on top.

> 3. I've not found Python to be particularly slow to build for such a large program. You may want to look at ccache if you want to speed you build process (I'm not using it on this computer and it's still quite fast, Make will only rebuild the appropriate object files already unless you explicitly clean them).

This guy has completely misunderstood the point: configuration is not build. If anything, ccache makes it even slower due to unnecessary I/O which is the slowest part even when using fastest NVMe.

> Or are there better options for a new build system for CPython, for example Meson? After all Meson is written in Python and used by large projects like X.org, Wayland, major parts of Gnome, systemd, and more.

This is a viable option. Again, basis on python is likely  a negative due to slowness of interpretation  but it's still light-years ahead autotools.

> PS: I find your phrasing and choice of words aggressive, condescending, patronizing. Please turn it down a nudge.

Ironically, this is the actual "aggressive, condescending, patronizing phrasing and choice of words" that needs to be "turned down a notch".

That being said, python's configuration and build are relatively fast compared to actual offenders. There is probably no immediate need for changing that but there is no obstacle to adding one to be used in parallel either.
